### PR TITLE
Watch the current template for changes

### PR DESCRIPTION
**Pull request by tbartelmess**
I ran into this use-case while writing a complex template.

The HFBinaryTemplateController now uses an FSEventStream to monitor the current binary template for changes.

It looks like there is already an issue for this https://github.com/HexFiend/HexFiend/issues/291

### CODE REVIEW

**Comment by kainjow** [comment]
I think it'd be cleaner to move this FSEvents code into its own class. Can you do this?

File: app/sources/HFBinaryTemplateController.m [hunk 1]
```diff
@@ -55,17 +55,31 @@ @interface HFBinaryTemplateController () <NSOutlineViewDataSource, NSOutlineView
...
 @implementation HFBinaryTemplateController
 
 - (instancetype)init {
     if ((self = [super initWithNibName:@"BinaryTemplateController" bundle:nil]) != nil) {
+        [self setupFSEventStream];
     }
     return self;
 }
 
+
```
Comment (kainjow) [inline app/sources/HFBinaryTemplateController.m:82]: unnecessary extra newline

**Comment by tbartelmess** [comment]
> I think it'd be cleaner to move this FSEvents code into its own class. Can you do this?

Sure. However it might be next week until I get to this.

**Comment by kainjow** [comment]
> > I think it'd be cleaner to move this FSEvents code into its own class. Can you do this?
> 
> Sure. However it might be next week until I get to this.

bump on this :)

**Comment by kainjow** [comment]
I used your changes to implement this slightly differently - https://github.com/HexFiend/HexFiend/commit/4128c583c526dc91be864055dad1fe638100a711

Thanks!

**Pull request closed without merging**